sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/JsonSerializable.java

```diff
@@ -526,9 +527,23 @@ private String toPrettyJson(Object object){
      * @return the POJO.
      */
     public <T> T toObject(Class<T> c) {
+        // TODO: We have to remove this if we do not want to support CosmosItemProperties anymore, and change all the
+        //  tests accordingly
+        if(CosmosItemProperties.class.isAssignableFrom(c)){
+            return (T) new CosmosItemProperties(this.toJson());
+        }
+        
         if (JsonSerializable.class.isAssignableFrom(c) || String.class.isAssignableFrom(c)
                 || Number.class.isAssignableFrom(c) || Boolean.class.isAssignableFrom(c)) {
-            throw new IllegalArgumentException("c can only be a POJO class or JSONObject");
+            return c.cast(this.get(Constants.Properties.VALUE));
+        }
+        if (List.class.isAssignableFrom(c)){
```

Comment:
Do we have tests for templatized List?
Any specific limitation around supporting `List<S>`?

Reply:
List<S> is supported right now, but we donot do any special processing for it, and List.class actually takes care types like List<S>, List<List<S> etc. I already have a test for List<List<Integer>>